src/main/java/com/textmessenger/controller/UserController.java

```diff
@@ -42,8 +53,28 @@ public ResponseEntity endpoint() {
   }
 
   @PostMapping("/user")
-  public ResponseEntity createUser(@RequestBody User user) {
-    return ResponseEntity.status(201).body(userService.createUser(user));
+  public ResponseEntity createUser(@Valid @RequestBody User user) {
+    User user1;
+    SimpleMailMessage email = new SimpleMailMessage();
+    if (userService.findUserByEmailOrLogin(user).get().size() == 0) {
+      user1 = userService.createUser(user);
+      TemporaryToken tempToken = new TemporaryToken();
+      tempToken.setToken(UUID.randomUUID().toString());
+      tempToken.setExpiryDate(new Date());
+      tempToken.setUser(user1);
+      temporaryTokenRepository.save(tempToken);
+      email.setTo(user1.getEmail());
+      email.setSubject("confirmation link to create account at Text Messanger application");
+      email.setText("http://localhost:3000/api/users/registered/" + tempToken.getToken());
+      emailService.sendEmail(email);
+      return ResponseEntity.ok().build();
+    }
+    return ResponseEntity.status(401).build();
+  }
+
+  @GetMapping("/registered/{token}")
+  public void enableUser(@PathVariable("token") String token) {
+    String s = userService.setUserIsEnabled(token);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unused "s" local variable. ([squid:S1481](https://sonarcloud.io/organizations/final_project_text_messenger/rules#rule_key=squid%3AS1481))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=Text_Messenger_Project&pullRequest=166&issues=AWWhHkYD3Qu1svVwBtp6&open=AWWhHkYD3Qu1svVwBtp6)

[ISSUE_KEY]: AWWhHkYD3Qu1svVwBtp6